eZ/Publish/Core/REST/Server/Controller/SessionController.php

```diff
@@ -92,8 +92,10 @@ public function createSessionAction(Request $request)
             // Already logged in with another user, this will be converted to HTTP status 409
             return new Values\Conflict();
         } catch (AuthenticationException $e) {
+            $this->authenticator->logout($request);
```

Comment:
Is the user authenticated when there is an authentication error ? I understand that the later case, where the user doesn't have enough permissions, leaves the user logged in, but I don't think it does, or should, when auth failed.